File: front/frontend-admin/app/api/LastExecution.ts
/// <reference path="api.d.ts" />

namespace api {
    'use strict';

    export interface LastExecution {

        "lastActualExecutionTime"?: Date;

        "lastCompletionTime"?: Date;

        "lastScheduledExecutionTime"?: Date;

        "status"?: string;
    }

}

File: backend/companies-queue/src/main/java/com/datapublica/companies/model/scheduler/LastExecution.java
package com.datapublica.companies.model.scheduler;

import java.util.Date;

import org.springframework.scheduling.TriggerContext;
import org.springframework.scheduling.support.SimpleTriggerContext;

/**
 *
 */
public class LastExecution {
    public Date lastActualExecutionTime;
    public Date lastCompletionTime;
    public Date lastScheduledExecutionTime;
    public String status;

    public TriggerContext asContext() {
        SimpleTriggerContext context = new SimpleTriggerContext();
        context.update(lastScheduledExecutionTime, lastActualExecutionTime, lastCompletionTime);
        return context;
    }
}
